bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -101,25 +102,46 @@ final class BagsManager(executor: ScheduledThreadPoolExecutor) {
       synchronized {
         val key = obs.getNodeKey
         state += key
-        executor.schedule(new Thread {
-          setPriority(Thread.NORM_PRIORITY - 1)
 
-          override def run(): Unit =
+        def bagsStatus(status: BagsStatus) = bagsStatuses.synchronized {
+          val oldState = bagsStatuses.get(key)
+          bagsStatuses += ((key, status))
+          notifyBagsStatusListeners(observation, oldState, Some(status))
+        }
+        def bagsClearStatus() = bagsStatuses.synchronized {
+          val oldState = bagsStatuses.get(key)
+          bagsStatuses -= key
+          notifyBagsStatusListeners(observation, oldState, None)
+        }
+
+        // The criteria to perform a BAGS lookup is that:
+        // 1. There are no manual primary targets in the primary guide group;
+        // 2a. This is not the initial call to enqueue (i.e. the call was triggered by a change to the TargetEnv); or
+        // 2b. The primary guide group has some active guide probe for which a BAGS search has not been done; and
+        // 3. The observation has not been executed.
+        def obsCtxFilter(obsCtx: ObsContext): Boolean =
+          !obsCtx.hasManualPrimary &&
+            (!initialEnqueue || obsCtx.isMissingSearch) &&
+            ObservationStatus.computeFor(obs) != ObservationStatus.OBSERVED
 
-          // If dequeue is false this means that (a) another task scheduled *after* me ended up
-          // running before me, so their result is as good as mine would have been and we're done;
-          // or (b) we don't care about that program anymore, so we're done.
-            if (dequeue(key, obs.getProgramID)) {
-              // Otherwise construct an obs context, verify that it's bagworthy, and go
-              ObsContext.create(obs).asScalaOpt.filter(o => !initialEnqueue || o.isEligibleForBags).foreach { ctx =>
+        // If dequeue is false this means that (a) another task scheduled *after* me ended up
+        // running before me, so their result is as good as mine would have been and we're done;
+        // or (b) we don't care about that program anymore, so we're done.
+        if (dequeue(key, obs.getProgramID)) {
+          // Otherwise construct an obs context, verify that it's bagworthy, and go
+          ObsContext.create(obs).asScalaOpt.filter(obsCtxFilter).foreach { ctx =>
+            bagsStatus(BagsStatus.Pending)
+            executor.schedule(new Thread {
```

Comment:
The reason I used a `Thread` instead of a `Runnable` is because a `Thread` allows for priority setting, which we wanted to ensure is used despite much discussion over whether or not it was relevant to make BAGS run at a lower priority and not interfere with the OT speed.